### PR TITLE
Review upload

### PR DESCRIPTION
**Pull request by vitalik12345678**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java [hunk 1]
```diff
@@ -17,36 +23,51 @@
...
-            Files.copy(inputStream, filePath, StandardCopyOption.REPLACE_EXISTING);
-        } catch (IOException ioe) {
-            throw new FileUploadException(String.format(FILE_UPLOAD_EXCEPTION, fileName));
-        }
+    @Override
+    public String uploadImage(String uploadDir, String fileName, MultipartFile multipartFile,Long id) {
+        saveFile(uploadDir, fileName, multipartFile);
 
         String actualPath = String.format("/%s/%s", uploadDir, fileName);
+        Optional<Club> optionalClub = clubRepository.findById(id);
```
Comment (vitalik12345678) [inline src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java:63]: DB add

File: src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java [hunk 2]
```diff
@@ -76,23 +97,99 @@ public void deleteImages(String urlLogo, String urlBackground, List<GalleryPhoto
...
+        uploadDir = uploadDir+"/"+ file.getOriginalFilename();
+        uploadDir = uploadDir.substring(uploadDir.indexOf("/"));
+
+        galleryPhoto.setUrl(uploadDir);
+        galleryRepository.save(galleryPhoto);
+
+        return "File changed successful";
+    }
+
+    private void saveFile(String uploadDir, String fileName, MultipartFile multipartFile) {
```
Comment (vitalik12345678) [inline src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java:132]: Method for save file on hard disk

File: src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java [hunk 3]
```diff
@@ -76,23 +97,99 @@ public void deleteImages(String urlLogo, String urlBackground, List<GalleryPhoto
...
+
+        try (InputStream inputStream = multipartFile.getInputStream()) {
+            Path filePath = uploadPath.resolve(fileName);
+            Files.copy(inputStream, filePath, StandardCopyOption.REPLACE_EXISTING);
+        } catch (IOException ioe) {
+            throw new FileUploadException(String.format(FILE_UPLOAD_EXCEPTION, fileName));
+        }
+    }
+
+    private void delete(String filePath){
```
Comment (vitalik12345678) [inline src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java:174]: Method for delete file from disk

File: src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java [hunk 4]
```diff
@@ -76,23 +97,99 @@ public void deleteImages(String urlLogo, String urlBackground, List<GalleryPhoto
...
-    public void deleteFile(String filePath) {
+    public String deleteFile(String filePath) {
+        delete(filePath);
+        GalleryPhoto galleryPhoto = galleryRepository.findByUrl(filePath);
+        galleryRepository.delete(galleryPhoto);
+        return filePath+" successfully deleted.";
+    }
+
+    @Override
+    public String updatePhoto(MultipartFile file,String filePath,String uploadDirectory) {
```
Comment (vitalik12345678) [inline src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java:111]: We saveing the file from disk after delete and changed url to bd

**Review by VictoriaAmazonka — CHANGES_REQUESTED** [review]
Post and request params….

**Pull request closed without merging**